### PR TITLE
Adding a general hook 'elementor/frontend/before_render' to element-base

### PR DESCRIPTION
**Pull request by iMaGd**
Handy hook for applying new attributes to the main wrapper for multiple elements with one single hook.

### CODE REVIEW

File: includes/base/element-base.php [hunk 1]
```diff
@@ -585,6 +585,17 @@ public function print_render_attribute_string( $element ) {
...
+		/**
+		 * Before frontend element render.
+		 *
+		 * Fires before Elementor element is rendered in the frontend.
+		 *
+		 * @since 2.1.0
+		 *
+		 * @param Element_Base $this The element.
+		 */
+		do_action( "elementor/frontend/before_render", $this );
```
Comment (KingYes) [inline includes/base/element-base.php]: Replace `"` => `'`.

File: includes/base/element-base.php [hunk 2]
```diff
@@ -585,6 +585,17 @@ public function print_render_attribute_string( $element ) {
 	public function print_element() {
 		$element_type = $this->get_type();
 
+		/**
+		 * Before frontend element render.
+		 *
+		 * Fires before Elementor element is rendered in the frontend.
+		 *
+		 * @since 2.1.0
```
Comment (KingYes) [inline includes/base/element-base.php]: Change `@since 2.1.0` to `@since 2.2.0`

**Review by KingYes — CHANGES_REQUESTED** [review]

File: includes/base/element-base.php (hunk 2 above)
Comment (M4Gd) [reply includes/base/element-base.php]: Sure!

**Comment by KingYes** [comment]
Thanks

**Review by KingYes — APPROVED** [review]

**Comment by M4Gd** [comment]
@KingYes Could you please add these changes in next upcoming update of Elementor? We prepared an update regarding these changes for our product and currently waiting for the changes to be available on new release of Elementor. Could you kindly instruct me on this matter?

**Comment by M4Gd** [comment]
Thanks for your response! Nevermind, we will find a solution on our own.

**Pull request merged**